### PR TITLE
Adding more breadcrumbs to diagnose SqlErrorQueryReturnedNoRows

### PR DESCRIPTION
**Pull request by bendk**
Based on the current logging, it seems like multiple write operations
are happening at the same time.  Added breadcrumbs around the
transaction begin/commit to test this theory further.

Also removed commented out SQL from `delete_visits_between_in_tx()`.

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[ac: android-components-branch-name]` and/or `[fenix: fenix-branch-name]` to the PR title.

### CODE REVIEW

**Comment by bendk** [comment]
When Mark and I discussed this last week, we also discussed removing the `delete places page` breadcrumb, since we were seeing lots of those in the DB log.  However, I didn't do that for 2 reasons:
  - When I went through more of the Sentry events, many of them didn't have that breadcrumb
  - The original theory was that these calls came from some bulk delete method and it would be better to just log the number of pages deleted.  However, after looking through the code more, I believe the top-level call is `bookmarks_delete()`, which inputs a single GUID.  It seems like the user is actually clicking on the trash icon for individual bookmarks.

File: components/places/src/storage/history.rs [hunk 1]
```diff
@@ -405,14 +413,18 @@ pub fn prune_destructively(db: &PlacesDb) -> Result<()> {
 }
 
 pub fn prune_older_visits(db: &PlacesDb) -> Result<()> {
-    let _tx = db.begin_transaction()?;
```
Comment (mhammond) [inline components/places/src/storage/history.rs:408]: oops - this was a bug that the tx wasn't committed, right?

**Review by mhammond — APPROVED** [review]
👍

File: components/places/src/storage/history.rs (hunk 1 above)
Comment (bendk) [reply components/places/src/storage/history.rs:408]: I believe it gets committed when `_tx` gets dropped.

Comment (mhammond) [reply components/places/src/storage/history.rs]: https://docs.rs/rusqlite/0.12.0/rusqlite/struct.Transaction.html#method.drop_behavior says "Transactions will roll back by default. Use commit method to explicitly commit the transaction, or use set_drop_behavior to change what happens when the transaction is dropped." and I can't see where we call `set_drop_behavior`.

**Pull request merged**